sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -1346,7 +1346,7 @@ def test_one_hot_encoder_sparse_deprecated():
 
 # deliberately omit 'OS' as an invalid combo
 @pytest.mark.parametrize(
-    "input_dtype, category_dtype", ["OO", "OU", "UO", "UU", "US", "SO", "SU", "SS"]
+    "input_dtype, category_dtype", ["OO", "OU", "UO", "UU", "SO", "SU", "SS"]
```

Comment:
The combination of `fit` input with type `U` and predefined categories as `S` lead to comparing an array of bytes to an array of strings.

In `_fit` we ended up with `cats = array([b'b', b'a'], dtype=object)` and `Xi = array(['b', 'a'], dtype='<U1')`, which then failed the `_check_unknown()` test. I think it makes sense that this happens. An array of bytes and an array of strings can't really be compared, without assuming an encoding.

A bit unsure what to do here. I don't like having to modify an existing test to "make it pass".

Reply:
I've added a test for this specific case, as well as code in the estimator that raises a particular message to help users figure out what happened. WDYT?

Reply:
+1 on my side.